### PR TITLE
Optimize/add static math functions

### PR DESCRIPTION
**Pull request by runer112**
Adds or optimizes static versions of many math functions. Most notably, implements the full suite of 64-bit integer (`long long`) support.

Partially addresses #276.

TODO:
- [x] This description.
- [x] Autotester config for math_test.
- [x] Confirm that no more integer math functions are missing.
- [x] Optimize static `short` division family of functions.
- [x] Verify all functions over many inputs.

### CODE REVIEW

File: src/std/shared/llneg_fast.src [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	ex	de, hl		; ude=UDE, uhl=UHL
+	ld	hl, 0		; uhl=0
+	sbc	hl, de		; uhl=-UDE-(UHL!=0)
+				;    =-UDEUHL>>24
+	ex	de, hl		; ude=-UDEUHL>>24, uhl=UDE
+	sbc	a, c		; a=-C-(UDEUHL!=0)
+				;  =-CUDEUHL>>48
+	ld	c, a		; c=-CUDEUHL>>48
+	ld	a, 0		; a=0
+	sbc	a, b		; a=-B-(CUDEUHL!=0)
```
Comment (runer112) [inline src/std/shared/llneg_fast.src:20]: > 11:01:34 <•PT_> Also, @Runner112, your llneg_fast can be optimized: line 19 and 20 can be replaced with "sbc a, a \ sub a, b" which yields the same result as "ld a, 0 \ sbc a, b" but is 1 byte shorter

File: src/std/shared/llneg_fast.src [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	public	__llneg_fast
+__llneg_fast:
+	push	de		; *spl=UDE
+	ex	de, hl		; ude=UHL, uhl=UDE
+	xor	a, a		; a=0
+	sbc	hl, hl		; uhl=0
+	sbc	hl, de		; uhl=-UHL
+	ex	(sp), hl	; uhl=UDE, *spl=-UHL
+	ex	de, hl		; ude=UDE, uhl=UHL
+	ld	hl, 0		; uhl=0
```
Comment (runer112) [inline src/std/shared/llneg_fast.src:12]: `ld l, a \ mlt hl`

**Comment by mateoconlechuga** [comment]
*bump*

File: src/makefile.mk [hunk 3]
```diff
@@ -39,7 +39,9 @@ CUSTOM_FILE_FILE ?= stdio_file.h
 DEPS ?=
 #----------------------------
 HAS_UPPERCASE_NAME ?= YES
-HAS_FLASH_FUNCTIONS ?= YES
+# Prefer static crt functions.
+# These will increase output size but are usually faster than those from TI-OS.
+STATIC_CRT ?= YES
```
Comment (runer112) [inline src/makefile.mk:44]: Do people want a different name/behavior for this?

Comment (mateoconlechuga) [reply src/makefile.mk:44]: Seems fine to me

**Comment by mateoconlechuga** [comment]
*poke* *poke*

**Comment by mateoconlechuga** [comment]
Might be worth also exploring what C/C++ functions need to be added as well for long long support.

**Comment by jacobly0** [comment]
```
Testing _llabs... 0.000000% failed in 146/223.505890/301 cycles minimum/average/maximum
Testing __lladd... 0.000000% failed in 241/241.498825/242 cycles minimum/average/maximum
Testing __lland... 95.303300% failed in 359/359.000000/359 cycles minimum/average/maximum
Check failed for test #0 with input:
	AF 7FFF     7F40 AF' (SP+00) 3FFFFE
	BC FEC0FF C0FE40 BC' (SP+03) BF963A
	DE E0FFC0 7FFF3F DE' (SP+06) 0040FF
	HL FFBF3F FE80BF HL' (SP+09) BFABFE
	IX 2D227F D3FFFD SP  (SP+12) 000000
	IY 80C080 000000 PC  (SP+15) 000400
and output:
	AF 3A3C     7F40 AF' (SP+00) BF963A
	BC FE80FE C0FE40 BC' (SP+03) 0040FF
	DE 0040C0 7FFF3F DE' (SP+06) BFABFE
	HL BF963A FE80BF HL' (SP+09) 000000
	IX 2D227F D40000 SP  (SP+12) 000400
	IY 80C080 3FFFFF PC  (SP+15) 000000
Testing __llrems... 37.529400% failed in 8482/9944.709816/10784 cycles minimum/average/maximum
Check failed for test #2 with input:
	AF 403F     FEBF AF' (SP+00) 3FFFFE
	BC BF7F40 00FE01 BC' (SP+03) FF0019
	DE C03F3F ACFE80 DE' (SP+06) FFBF7F
	HL 40FF80 40FE40 HL' (SP+09) 80EBBF
	IX 804880 D3FFFD SP  (SP+12) 000000
	IY C00080 3FFFFF PC  (SP+15) 000400
and output:
	AF 407D     FEBF AF' (SP+00) B1A803
	BC 000D81 000000 BC' (SP+03) FCD097
	DE 40FB7F 001440 DE' (SP+06) 94FFFF
	HL 490FCE 000D81 HL' (SP+09) 000000
	IX 804880 D40000 SP  (SP+12) 000400
	IY C00080 3FFFFF PC  (SP+15) 000000
Testing __lldivs... 25.057100% failed in 8720/10023.872314/10704 cycles minimum/average/maximum
Check failed for test #2 with input:
	AF 403F     FEBF AF' (SP+00) 3FFFFE
	BC BF7F40 00FE01 BC' (SP+03) FF0019
	DE C03F3F ACFE80 DE' (SP+06) FFBF7F
	HL 40FF80 40FE40 HL' (SP+09) 80EBBF
	IX 804880 D3FFFD SP  (SP+12) 000000
	IY C00080 3FFFFF PC  (SP+15) 000400
and output:
	AF 402A     FEBF AF' (SP+00) B1A803
	BC 940000 000000 BC' (SP+03) FCD097
	DE 032F68 001440 DE' (SP+06) 94FFFF
	HL 4E57FD 000D81 HL' (SP+09) 000000
	IX 804880 D40000 SP  (SP+12) 000400
	IY C00080 3FFFFF PC  (SP+15) 000000
Testing __lldivu... 0.000000% failed in 9080/9981.399369/10010 cycles minimum/average/maximum
Testing __llmulu... 0.000000% failed in 6560/9008.892639/11379 cycles minimum/average/maximum
Testing __llneg... 0.000000% failed in 161/161.000000/161 cycles minimum/average/maximum
Testing __llnot... 0.000000% failed in 141/141.000000/141 cycles minimum/average/maximum
Testing __llor... 95.322800% failed in 359/359.000000/359 cycles minimum/average/maximum
Check failed for test #0 with input:
	AF 7FFF     7F40 AF' (SP+00) 3FFFFE
	BC FEC0FF C0FE40 BC' (SP+03) BF963A
	DE E0FFC0 7FFF3F DE' (SP+06) 0040FF
	HL FFBF3F FE80BF HL' (SP+09) BFABFE
	IX 2D227F D3FFFD SP  (SP+12) 000000
	IY 80C080 000000 PC  (SP+15) 000400
and output:
	AF 3F2C     7F40 AF' (SP+00) BF963A
	BC FEEBFF C0FE40 BC' (SP+03) 0040FF
	DE E0FFFF 7FFF3F DE' (SP+06) BFABFE
	HL FFBF3F FE80BF HL' (SP+09) 000000
	IX 2D227F D40000 SP  (SP+12) 000400
	IY 80C080 3FFFFF PC  (SP+15) 000000
Testing __llremu... 0.000000% failed in 9008/9909.399369/9938 cycles minimum/average/maximum
Testing __llshl... 0.000000% failed in 162/1242.170493/2987 cycles minimum/average/maximum
Testing __llshrs... 0.000000% failed in 150/4170.104475/10587 cycles minimum/average/maximum
Testing __llshru... 0.000000% failed in 154/4189.982432/10612 cycles minimum/average/maximum
Testing __llsub... 0.000000% failed in 251/251.000000/251 cycles minimum/average/maximum
Testing __llxor... 96.702200% failed in 359/359.000000/359 cycles minimum/average/maximum
Check failed for test #0 with input:
	AF 7FFF     7F40 AF' (SP+00) 3FFFFE
	BC FEC0FF C0FE40 BC' (SP+03) BF963A
	DE E0FFC0 7FFF3F DE' (SP+06) 0040FF
	HL FFBF3F FE80BF HL' (SP+09) BFABFE
	IX 2D227F D3FFFD SP  (SP+12) 000000
	IY 80C080 000000 PC  (SP+15) 000400
and output:
	AF 052C     7F40 AF' (SP+00) BF963A
	BC FE6B01 C0FE40 BC' (SP+03) 0040FF
	DE E0BF3F 7FFF3F DE' (SP+06) BFABFE
	HL 402905 FE80BF HL' (SP+09) 000000
	IX 2D227F D40000 SP  (SP+12) 000400
	IY 80C080 3FFFFF PC  (SP+15) 000000
```

**Comment by runer112** [comment]
> [Test results.](https://gist.github.com/jacobly0/4637158444eca148a330b6b9100b445d)_ (too large to embed)_

Thanks for properly testing this. I've fixed all of these failures.

More testing, please.

**Comment by jacobly0** [comment]
[More testing](https://github.com/jacobly0/ez80sf/runs/5460422445?check_suite_focus=true#step:3:80).

**Review by jacobly0 — APPROVED** [review]
LGTM, [all tests](https://github.com/jacobly0/ez80sf/runs/5617678272?check_suite_focus=true#step:3:80) are passing.

**Pull request merged**